### PR TITLE
Refactor repository to use distinct physics module 

### PR DESCRIPTION
**Pull request by joshuaRiefman**
## Overview
1. All physics and environment models have been moved into a new project, ``ubc-solar-physics``, which has been added as a dependency to ``Simulation``.
2. ``Poetry`` has been injected as a package and dependency manager. 
3. ``display_speeds.py`` is a new script allowing for interactive viewing of optimized speed outputs.

## Fixes
1. Bokeh has been updated to latest version, and API calls consequently updated. Also, ``.html`` files now get placed into a ``html/`` directory.
2. ``GeneticOptimization`` fitness penalization for killing the battery has been maximized to improve convergence.
3. ``run_simulation`` documentation has been improved
4.  ``update_environment`` has been fixed to factor in timezone offset between local and destination locations.

## Refactoring
1. As all Rust code was relevant to physics models only, Rust build configuration has been removed.
2. All constants have been updated for BrightSide
5. ``cache`` module now has dedicated methods for storing and loading in a standard manner.
6. ``README.md`` has been overhauled, and ``USAGE_GUIDE.md`` added to explain how to use our scripts.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,13 +8,16 @@ This document contains information on how to get setup with the Simulation frame
...
 For more detailed information on the inner workings of the simulation please refer to the [wiki](https://github.com/UBC-Solar/Simulation/wiki).
 
+For instructions on how to use the scripts provided, see our [usage guide](simulation/cmd/USAGE_GUIDE.md).
+
 ## Getting started
 
 ### Prerequisites
 
-- Python 3.8 or above (https://www.python.org/downloads/)
+- Python 3.9 or above (https://www.python.org/downloads/)
```
Comment (lambpasta) [inline README.md:17]: image

Python 3.9 does not seem to be supported for me

Comment (tamzeedq) [reply README.md:17]: worked for me (Python 3.9.13) maybe it's a local issue? @lambpasta

Comment (joshuaRiefman) [reply README.md:17]: I have fixed the README to state Python 3.10 is required

File: README.md [hunk 2]
```diff
@@ -48,106 +54,58 @@ NOTE: Any Python version before 3.8 is not supported so please make sure that yo
...
-- To install the simulation package, run the following:
-
-    ```bash
-    pip install .
-    ```
+- Simulation uses `poetry` to manage dependencies. First, we will create a virtual environment. _Note: the second line is platform-dependent, and may be different for you._
+
+  ```bash
+  python3 -m venv .venv
+  source .venv/bin/activate
```
[the remark this answers is not in the document]
Comment (joshuaRiefman) [reply README.md:61]: Yep, thank you! I have added it to the README

File: README.md [hunk 3]
```diff
@@ -48,106 +54,58 @@ NOTE: Any Python version before 3.8 is not supported so please make sure that yo
...
+
+  - Now, use `poetry` to install dependencies.
+  
+  ```bash
+  poetry install
+  ```
+  
+  - Finally, compile the Rust libraries.
+  ```bash
+  maturin develop
```
[the remark this answers is not in the document]
Comment (joshuaRiefman) [reply README.md:73]: Yes that is an oversight, `maturin` is no longer a dependency of `Simulation` as all of the Rust code is in the `physics` module.

File: simulation/optimization/genetic.py [hunk 4]
```diff
@@ -469,17 +469,17 @@ def fitness(self, ga_instance, solution, solution_idx) -> float:
 
         # If Simulation did not complete successfully (SOC dropped below 0) then return the distance when that occurred.
         distance_travelled_real = distance_travelled if self.model.was_successful() \
-            else self.model.get_distance_before_exhaustion()
+            else 0.0
 
         self.did_finish_race = True if distance_travelled_real == distance_travelled else False
```
[the remark this answers is not in the document]
Comment (joshuaRiefman) [reply simulation/optimization/genetic.py:474]: Yep that's true, I'll remove the field

File: simulation/optimization/genetic.py [hunk 5]
```diff
@@ -469,17 +469,17 @@ def fitness(self, ga_instance, solution, solution_idx) -> float:
...
-
-        # combined_fitness = distance travelled (km) / time_taken (days) = distance travelled per day
-        combined_fitness = distance_scaled / (time_taken_scaled / 86400)
+        # # distance_travelled is scaled such that optimization REALLY prioritizes finishing the race
+        # distance_scaled = distance_travelled_real * self.fitness_sigmoid(distance_travelled_real)
+        # # optimization really likes even small reductions in time taken
+        # time_taken_scaled = time_taken / self.fitness_amplifier(time_taken)
+        #
+        # # combined_fitness = distance travelled (km) / time_taken (days) = distance travelled per day
+        # combined_fitness = distance_scaled / (time_taken_scaled / 86400)
```
[the remark this answers is not in the document]
Comment (joshuaRiefman) [reply simulation/optimization/genetic.py:482]: Agreed

**Review by tamzeedq — APPROVED** [review]

**Pull request merged**